tests/topology_test.go

```diff
@@ -837,3 +838,44 @@ func TestQueryMetadata(t *testing.T) {
 
 	RunTest(t, test)
 }
+
+//TestUserMetadata tests user metadata functionality
+func TestUserMetadata(t *testing.T) {
+	test := &Test{
+		setupCmds: []helper.Cmd{
+			{"ovs-vsctl add-br br-umd", true},
+		},
+
+		tearDownCmds: []helper.Cmd{
+			{"ovs-vsctl del-br br-umd", true},
+		},
+
+		check: func(c *TestContext) error {
+			gh := c.gh
+
+			prefix := "g"
+			if !c.time.IsZero() {
+				prefix += fmt.Sprintf(".Context(%d)", common.UnixMillis(c.time))
+			}
+
+			umd := api.NewUserMetadata(prefix+".V().Has('Name', 'br-umd', 'Type', 'ovsbridge')", "testKey", "testValue")
```

Comment:
These 2 lines should be in 'setupFunction', not in 'check' otherwise they could be executed multiple times